internal/resource/url.go

```diff
@@ -255,9 +255,6 @@ func (f *Fetcher) FetchFromHTTP(u url.URL, dest *os.File, opts FetchOptions) err
 // FetchFromDataURL writes the data stored in the dataurl u into dest, returning
 // an error if one is encountered.
 func (f *Fetcher) FetchFromDataURL(u url.URL, dest *os.File, opts FetchOptions) error {
-	if opts.Compression != "" {
```

Comment:
Should we also have tests for this change, similar to upstream?

Reply:
I also considered importing the tests but:
* there is no automated CI for the `flatcar-master`
* tests will be included when we'll switch to `main` branch

Reply:
Well, at least we could run them locally. I don't mind setting up a quick GH action as well.

If that's a matter of just cherry-picking entire commit, why not have it.